### ISSUE TITLE
FluxMaps.slice_by_coord behaviour for different Axis types

### ISSUE THREAD

**Issue by AtreyeeS** [gammapy/gammapy#5003]
The current behaviour of `FluxPoints.slice_by_coord` is rather confusing while slicing along the time axis. 
eg: 

```
from gammapy.estimators import FluxPoints
lc_1d = FluxPoints.read(
    "$GAMMAPY_DATA/estimators/pks2155_hess_lc/pks2155_hess_lc.fits", format="lightcurve"
)
l0 = lc_1d.slice_by_coord({"time":slice(lc_1d.geom.axes["time"].edges[0], lc_1d.geom.axes["time"].edges[1])})
l1 = lc_1d.slice_by_coord({"time":slice(lc_1d.geom.axes["time"].edges[1], lc_1d.geom.axes["time"].edges[2])})
```

`l0.geom` has 0 bins in time and `l0.geom.axes["time"]` raises an Error. `l1` does not.
If the `time` axis is non-contiguous, both `l0` and `l1` lead to an error

This is because `TimeMapAxis.coord_to_pix` assumes an open interval on the first edge, and a closed on the last edge, which is OK in itself, but leads to different behaviour from the `MapAxis`. For example, 

```
start = np.linspace(0,10,5) * u.day
stop = np.linspace(0.1,10.1,5) * u.day
t_ref = Time(55555, format="mjd")
gti = GTI.create(start, stop, reference_time=t_ref)
time_ax = TimeMapAxis.from_gti(gti)
time_ax.coord_to_idx(time_ax.edges_max), time_ax.coord_to_idx(time_ax.edges_min)
```

results `(array([0, 1, 2, 3, 4]), array([-1, -1, -1, -1, -1]))`

while 

```
time_ax2 = TimeMapAxis.from_time_bounds(Time(55555,format='mjd'), Time(55558,format='mjd'),
                                        nbin=5)
time_ax2.coord_to_idx(time_ax2.edges_max), time_ax2.coord_to_idx(time_ax2.edges_min)
```
results `(array([0, 1, 2, 3, 4]), array([-1,  0,  1,  2,  3]))`

(For a `MapAxis`, the behaviour is as expected - no invalid indices
```
energy_axis = MapAxis.from_energy_bounds(
         "1 TeV", "10 TeV", nbin=5, name="energy_true"
     )
energy_axis.coord_to_idx(energy_axis.edges_max), energy_axis.coord_to_idx(energy_axis.edges_min)
```
results `(array([1, 2, 3, 4, 5]), array([0, 1, 2, 3, 4]))`
)

I think the time and energy cases should behave similarly, no?

**Comment by cgalelli** [comment]
Thanks for pointing out the issue @AtreyeeS, I agree that the behavior should be the same between `MapAxis` and `TimeMapAxis`, unless I am missing some point. If there is no problem with that, I could modify the `TimeMapAxis.coord_to_pix` as the `MapAxis` behavior seems to be the intended one.

**Comment by AtreyeeS** [comment]
I don't know if the fix should go in `coord_to_pix` or in `FluxMaps.slice_by_coord`, but the final behaviour should be similar to `MapDataset.slice_by_energy` for any given axis

**Comment by cgalelli** [comment]
This problem seems to be quite deep due to the differences between `MapAxis` and `TimeMapAxis`. Adding a `group_table` to `TimeMapAxis` is complicated because `node_type` is always `"interval"` in this case, while the method requires `node_type="edges"`. A quicker change would be using closed intervals instead of open, but this would shift quite a bit the situation on many levels.

**Comment by registerrier** [comment]
Is this solved by #5035? Can we close?

**Comment by AtreyeeS** [comment]
Addressed in https://github.com/gammapy/gammapy/pull/5035
